store.rb

```diff
@@ -35,14 +59,25 @@ def session_id
   def stop()
     @log.warn "stopping the store"
     Process.kill("TERM", Process.pid) unless $EXIT
+    @root_watcher.unsubscribe unless @root_watcher.nil?
```

Comment:
the `Process.kill` is a bit strange here.  I think we usually find other ways to shut ourselves down.